File: app/src/main/java/com/faciee/cti/valbastrelu/eticket/util/room/DateConverter.java
package com.faciee.cti.valbastrelu.eticket.util.room;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

public class DateConverter {

	@TypeConverter
	public Date fromTimeStamp(long value){
		return (value == 0 ? null : new Date(value));
	}
	
	@TypeConverter
	public Long dateToTimeStamp(Date date){
		if (date == null) return null;
		else return date.getTime();
	}
}

File: app/src/main/java/com/faciee/cti/valbastrelu/eticket/room/converter/DateConverter.kt
package com.faciee.cti.valbastrelu.eticket.room.converter

import androidx.room.TypeConverter
import java.util.*

class DateConverter {
	@TypeConverter
	fun fromTimeStamp(value: Long): Date? {
		return if (value == 0L) null else Date(value)
	}

	@TypeConverter
	fun dateToTimeStamp(date: Date?): Long? {
		return date?.time
	}
}